### PR TITLE
Rebase: Fix at_exit hook to cooperate with Minitest

### PR DESCRIPTION
**Pull request by PragTob**
This is me trying to resurrect #756

### CODE REVIEW

**Comment by PragTob** [comment]
@adam12 if you get to try this out to see if I didn't accidentally break the fix I'd be very happy :) But the feature works so I hope for the best :crossed_fingers:

**Comment by adam12** [comment]
Looks great. I'm going to give it a try sometime this week.

I never could figure out why the coverage changed on #756 between Ruby versions.

**Comment by PragTob** [comment]
@adam12 ah I squashed all the commits, basically in the `test_helper.rb` of the `begin .. end` the begin was marked as relevant in 2.5 but not the others which I consider a coverage bug. I solved it by ignoring `test_helper.rb` (not interested in its coverage anyhow).

Let me know what you find once you try it out :crossed_fingers: And thanks for your help with this :green_heart:

**Comment by PragTob** [comment]
@adam12 :wave: Do you know if you'll get to trying this out? No pressure, would just be nice since you had the initial problems. I'm looking to merge it some time this week.

**Comment by adam12** [comment]
> Do you know if you'll get to trying this out?

_Looks_ to be functioning correctly here. At least to say, I get coverage report at the end of the test run and not immediately before the tests start running as was happening previously.

**Comment by PragTob** [comment]
@adam12 thanks for trying it out! Will merge :grin:

File: lib/simplecov.rb [hunk 1]
```diff
@@ -189,6 +189,14 @@ def exit_status_from_exception
       end
     end
 
+    def at_exit_behvior
```
Comment (Yoshiji) [inline lib/simplecov.rb]: typo at `behvior`? `behavior`?

Comment (PragTob) [reply lib/simplecov.rb]: good catch, thank you!

**Pull request merged**